accepted/00075-live-patching-enhancements.md

```diff
@@ -0,0 +1,189 @@
+
+Below are some examples of such sets of filters:
+
+**Example A:** Live patching project with a single filter to exclude regular kernel packages newer than a specific version
+ - `DENY erratum contains package with epoch/version/release greater than kernel-default <EVR> (package_nevr)`
+
+**Example B:** Live patching project using `ALLOW` overrides
+ - `DENY errata containing "update for the Linux Kernel" in synopsis`
+ - `ALLOW errata containing "Live Patch" in synopsis` *- an override to the first rule*
+ - `ALLOW a specific patch containing current kernel version (via advisory name)` *- an override to the first rule*
+
+**Example C:** Live patching project using reboot flag
+ - `DENY errata containing reboot_suggested (keyword)`
+ - `ALLOW errata containing "Live Patch" in synopsis` *- an override to the first rule*
+ - `ALLOW a specific patch containing current kernel version (via advisory name)` *- an override to the first rule*
+
+### CLM filter wizards
+
+The process of live patching can be made more user-friendly using "CLM filter wizards". The wizards are alternatives to the regular filter creation dialogs that provide a tailored interaction to create multiple filters that serve a single purpose. These dialogs can be accessed via the "Use wizard" button in the filter creation dialog. This concept can be extended to other use cases and workflows in the future.
+
+When completed, the wizards will create multiple filters that look like the examples mentioned in the section above to set up a project with a specific purpose. Since live patching channels can be composed using the existing filter types, no additional filter type needs to be implemented.
+
+An advantage of this approach is that since the workflow is defined with a set of regular filters, advanced users can modify these filters to achieve finer control over the project.
+
+A wizard can be applied on a project at any time, adding filters as described. Multiple wizards can be applied to a single project and it's the user's responsibility to make sure the created filters do not conflict with each other.
+
+A wizard can be applied independently from a project. In that case, the filters will be created without being added to a specific project. These filters can be used in any project later on.
+
+In scope of this RFC, two different wizards are proposed. The process of creating the filters using these wizards are described below.
+
+#### Live patching for a system
+
+ 1. Input a label prefix for the resulting filters. If the wizard is accessed from the project view, prefill this field with the project label
+ 2. Select a SLES client from a combobox
+ 3. Select a kernel version from a dropdown (current kernel version of the system is preselected)
```

Comment:
The purpose of the system selection here is just to provide the user a way to say "I want my live patching based on this system's kernel version" (as opposed to looking for the version and adding the filters manually). It's useful because one of the most common use cases is that the user wants to setup live patching based on the currently installed kernel version of some of their systems.

> How does this look when there are multiple kernel version available in the same channel? I. e. are we sure this will not define filters to be used with a different kernel version?

The correct approach for the user would be to make sure all the systems assigned to the project have the same kernel version installed. If they have systems with older versions, they would still see the updates up to the picked version. On the other hand, if there are systems with newer versions than the one chosen, **those ones will have neither real updates nor the live-patching updates.**

Basically, a live-patching project will target a specific kernel version and therefore all the systems assigned must respect that.

Imagine a distribution that has kernel versions 1, 2 and 3. When you setup a live-patching project targeting version 2, you'll have the following packages in the channels:
![File_000](https://user-images.githubusercontent.com/1103552/105701313-6bedf280-5f0a-11eb-9708-7fd95e3d7391.png)